src/spaces/nothing.jl

```diff
@@ -3,17 +3,18 @@ This file implements the "agent-space interaction API" for `nothing`, ie
 no space type
 =#
 
-function kill_agent!(agent::A, model::ABM{Nothing,A}) where {A<:AbstractAgent}
-    delete!(model.agents, agent.id)
+function kill_agent!(agent::A, model::ABM{Nothing,A,Dict{Int,A}}) where {A<:AbstractAgent}
+    remove_agent_from_model!(agent, model)
 end
 
 function add_agent!(agent::A, model::ABM{Nothing,A}) where {A<:AbstractAgent}
     add_agent_pos!(agent, model)
 end
 
 function add_agent_pos!(agent::A, model::ABM{Nothing,A}) where {A<:AbstractAgent}
-    model[agent.id] = agent
-    return model[agent.id]
+    add_agent_to_model!(agent, model)
+    model.maxid[] < agent.id && (model.maxid[] = agent.id)
+    return agent
```

Comment:
Same like before: the only functions that should need special methods for add/remove agents are the `add_agent_to_model!`. Everything downstream should "just worK" without caring about the model, by simply calling the `add_agent_to_model!` etc. functions.

Reply:
I'm not sure I understand your comment. Currently the `add_agent_pos!` functions differ in dispatch because one of them adds the agent to the space, and the other has no space to add it to, so doesn't. Are you saying this should happen at the `add_agent_to_model!` level? i.e.
`add_agent_to_model!` dispatched on `ABM{Nothing, A}` vs `ABM{AbstractSpace, A}` to conditionally include `add_agent_to_space!` and there should only be *one* location in source code where `add_agent_pos!` is defined?